evalml/tests/pipeline_tests/test_pipeline_utils.py

```diff
@@ -856,59 +857,96 @@ def test_generate_pipeline_example(
     X_y_regression,
     ts_data,
 ):
-    path = os.path.join(str(tmpdir), "train.csv")
     if automl_type == ProblemTypes.BINARY:
         X, y = X_y_binary
     elif automl_type == ProblemTypes.MULTICLASS:
         X, y = X_y_multi
     elif automl_type == ProblemTypes.REGRESSION:
         X, y = X_y_regression
-    elif (
-        automl_type == ProblemTypes.TIME_SERIES_MULTICLASS
-        or automl_type == ProblemTypes.TIME_SERIES_BINARY
-    ):
-        X, _, y = ts_data(problem_type=automl_type)
-    else:
+    elif is_time_series(automl_type):
         X, _, y = ts_data(problem_type=automl_type)
 
-    from evalml import AutoMLSearch
-
-    aml = AutoMLSearch(
-        X_train=X,
-        y_train=y,
-        problem_type=automl_type,
-        optimize_thresholds=False,
-        max_time=1,
-        max_iterations=5,
-        problem_configuration={
+    problem_configuration = (
+        {
             "time_index": "date",
             "gap": 1,
             "max_delay": 1,
             "forecast_horizon": 3,
         }
         if is_time_series(automl_type)
-        else None,
+        else None
+    )
+
+    import featuretools as ft
+
+    from evalml import AutoMLSearch
+    from evalml.preprocessing import split_data
+
+    X_train, X_test, y_train, y_test = split_data(
+        X,
+        y,
+        problem_type=automl_type,
+        test_size=0.2,
+    )
+
+    X_train = pd.DataFrame(X_train)
+    X_train.columns = X_train.columns.astype(str)
+    es = ft.EntitySet()
```

Comment:
I can split the DFS/non-DFS case as well but I thought that the DFS case covers all the edge cases already so I kept it like this for now. LMK what you all think.